ouroboros-network/src/Ouroboros/Network/Diffusion.hs

```diff
@@ -140,47 +127,61 @@ runM
        , Eq extraCounters
        , Exception exception
        )
-    => -- | interfaces
-       Interfaces ntnFd ntnAddr ntnVersion ntnVersionData
-                  ntcFd ntcAddr ntcVersion ntcVersionData
-                  resolver resolverError
-                  extraState extraFlags extraPeers extraAPI m
+    => -- | arguments
+       Arguments extraState extraDebugState extraFlags
+                 extraPeers extraAPI extraChurnArgs
+                 extraCounters exception resolver
+                 resolverError m ntnFd
+                 ntnAddr ntnVersion ntnVersionData
+                 ntcAddr ntcVersion ntcVersionData
+      -- | interfaces
+    -> Interfaces ntnFd ntnAddr ntcFd ntcAddr
+                  resolver resolverError m
```

Comment:
I'd keep the original order.  `Interfaces` are the lowest-level API, so it makes sense to have them passed first, then `Traces`, `Arguments` followed by `Configuration`.